activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java

```diff
@@ -65,4 +74,31 @@ public IdentityManagementService identityManagementService(KeycloakClient keyclo
         return new KeycloakManagementService(keycloakClient);
     }
 
+    @Bean
+    public CaffeineCache groupRoleMappingCache() {
+        return new CaffeineCache("groupRoleMapping",
+                                 Caffeine.newBuilder()
+                                     .expireAfterWrite(Duration.parse(cacheExpireAfterWrite))
+                                     .maximumSize(cacheMaxSize)
+                                     .build());
+    }
+
+    @Bean
+    public CaffeineCache userRoleMappingCache() {
+        return new CaffeineCache("userRoleMapping",
+                                 Caffeine.newBuilder()
+                                     .expireAfterWrite(Duration.parse(cacheExpireAfterWrite))
+                                     .maximumSize(cacheMaxSize)
+                                     .build());
+    }
+
+    @Bean
+    public CaffeineCache userGroupsCache() {
+        return new CaffeineCache("userGroups",
+                                 Caffeine.newBuilder()
+                                     .expireAfterWrite(Duration.parse(cacheExpireAfterWrite))
+                                     .maximumSize(cacheMaxSize)
+                                     .build());
+    }
```

Comment:
Why are there differente cache If they use the same configuration parameters?

Reply:
caches are different because they are storing diffrent objects and they are using diffrerent keys